lib/json-schema/json-schema-validator.ts

```diff
@@ -88,8 +92,44 @@ export class JsonSchemaValidator implements IJsonSchemaValidator {
 		return propertyType;
 	}
 
-	private getValidationErrors(data: IProjectData): IStringDictionary {
-		var validationSchema = this.tryResolveValidationSchemaCore(data.Framework);
+	public validatePropertyUsingBuildSchema(propertyName: string, propertyValue: string): void {
+		var invalidPlatforms: IStringDictionary = Object.create(null);
+		var properties = Object.create(null);
+		properties[propertyName] = propertyValue;
+		_.each(this.$mobileHelper.platformNames, platformName => {
+			var validationErrors = this.getBuildSchemaValidationErrors(properties, platformName);
+			if(_.keys(validationErrors).length !== 0 && validationErrors[propertyName]) {
+				invalidPlatforms[platformName] = validationErrors[propertyName];
+			};
+		});
+
+		if(_.keys(invalidPlatforms).length === this.$mobileHelper.platformNames.length) {
+			this.$errors.fail("The validation of provided %s property failed with following errors:\n", propertyName, _.values(invalidPlatforms).join("\n"));
```

Comment:
also you have only one %s but there are two values that you want to include in the error message.